lib/lints/avoid_unnecessary_setstate/avoid_unnecessary_set_state_rule.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:solid_lints/lints/avoid_unnecessary_setstate/visitor/avoid_unnecessary_set_state_visitor.dart';
+import 'package:solid_lints/models/rule_config.dart';
+import 'package:solid_lints/models/solid_lint_rule.dart';
+
+/// A rule which warns when setState is called inside initState, didUpdateWidget
+/// or build methods and when it's called from a sync method that is called
+/// inside those methods.
+class AvoidUnnecessarySetStateRule extends SolidLintRule {
+  /// The [LintCode] of this lint rule that represents
+  /// the error whether we use setState in inappropriate way.
```

Comment:
I think it is not a `LintCode`, but a lint name, so we should not use a `LintCode` reference in the documentation.